### PR TITLE
implemented executer service to send the large chunk in threads

### PR DESCRIPTION
**Pull request by vitiwari**
@gokumar 
Implemented to send the events in multiple threads(Thread pool 20) with 30 events in each thread.

### CODE REVIEW

File: src/main/java/com/bmc/truesight/saas/remedy/integration/impl/CallableBulkEventHttpClient.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        int retryCount = 0;
+        while (!isSuccessful && retryCount <= this.configuration.getRetryConfig()) {
+            httpClient = HttpClientBuilder.create().build();
+            HttpPost httpPost = new HttpPost(this.configuration.getTsiEventEndpoint());
+            httpPost.addHeader("Authorization", "Basic " + encodeBase64("" + ":" + this.configuration.getTsiApiToken()));
+            httpPost.addHeader("Content-Type", "application/json");
+            httpPost.addHeader("accept", "application/json");
+            httpPost.addHeader("User-Agent", "RemedyScript");
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonInString = null;
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/saas/remedy/integration/impl/CallableBulkEventHttpClient.java]: I guess, you can initialize httpPost outside loop and reuse the same object every time with new entity

what say?

File: src/main/java/com/bmc/truesight/saas/remedy/integration/impl/EventIngestionAsync.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+import com.bmc.truesight.saas.remedy.integration.beans.Configuration;
+import com.bmc.truesight.saas.remedy.integration.beans.Error;
+import com.bmc.truesight.saas.remedy.integration.beans.Result;
+import com.bmc.truesight.saas.remedy.integration.beans.Success;
+import com.bmc.truesight.saas.remedy.integration.beans.TSIEvent;
+import com.bmc.truesight.saas.remedy.integration.exception.BulkEventsIngestionFailedException;
+import com.bmc.truesight.saas.remedy.integration.util.Constants;
+
+public class EventIngestionAsync {
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/saas/remedy/integration/impl/EventIngestionAsync.java]: Are we using this class yet?

**Review by gokumar — CHANGES_REQUESTED** [review]

File: src/main/java/com/bmc/truesight/saas/remedy/integration/impl/EventIngestionAsync.java (hunk 2 above)
Comment (vitiwari) [reply src/main/java/com/bmc/truesight/saas/remedy/integration/impl/EventIngestionAsync.java]: No, You want me to remove it ?

Comment (gokumar) [reply src/main/java/com/bmc/truesight/saas/remedy/integration/impl/EventIngestionAsync.java]: Please go ahead.

**Review by gokumar — APPROVED** [review]

**Pull request merged**